src/confluent_kafka/avro/load.py

```diff
@@ -16,42 +16,51 @@
 #
 
 
+from typing import TYPE_CHECKING
 from confluent_kafka.avro.error import ClientError
 
-
-def loads(schema_str):
-    """ Parse a schema given a schema string """
-    try:
-        return schema.parse(schema_str)
-    except SchemaParseException as e:
-        raise ClientError("Schema parse failed: %s" % (str(e)))
-
-
-def load(fp):
-    """ Parse a schema from a file path """
-    with open(fp) as f:
-        return loads(f.read())
-
-
-# avro.schema.RecordSchema and avro.schema.PrimitiveSchema classes are not hashable. Hence defining them explicitly as
-# a quick fix
-def _hash_func(self):
-    return hash(str(self))
-
-
 try:
+    # FIXME: Needs https://github.com/apache/avro/pull/1952
+    # pip install git+https://github.com/apache/avro#subdirectory=lang/py
```

Comment:
Avro master supports types, but they haven't released one yet that does. This was fixed in November 2022, so hopefully soon.